src/storage/v3/shard_rsm.cpp

```diff
@@ -325,7 +337,7 @@ std::optional<msgs::ExpandOneResultRow> GetExpandOneResult(
   const auto primary_key = ConvertPropertyVector(std::move(src_vertex.second));
   auto v_acc = acc.FindVertex(primary_key, View::NEW);
 
-  auto source_vertex = FillUpSourceVertex(v_acc, req, src_vertex);
+  auto source_vertex = FillUpSourceVertex(v_acc, req);
```

Comment:
What if instead of doing a `PropertyValue` -> `Value` conversion in `FillUpSourceVertex`, we just copy the primary key of the vertex on line 337 instead of moving it? Then it should be possible to use the passed in value to populate the result?

`FillUpSourceVertex` is already a bit tricky, because it only populates the vertex id and secondary labels. Instead we could populate the vertex id from the value passed in, and then the `FillUpSourceVertex` function could be transformed to `FillUpSourceVertexSecondaryLabels`. This way we don't have to pass the vertex id into a function and the function also return something that is complete on its own level.